From https://github.com/areaDetector/ADSupport/pull/41
From 0ecb38c9b8981b4e92c2ebbdff5503052a5c0ba9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=89rico=20Nogueira?= <erico.rolim@lnls.br>
Date: Thu, 3 Aug 2023 08:16:00 -0300
Subject: [PATCH] Fix nanohttp_stream build with external libxml2.

Commit 2e167d2bd5a002b079589abb32b4e454e234355e defined IN_LIBXML
unconditionally inside nanohttp_stream.c, which, when building on Linux
with external libxml2, leads to multiple linking errors, since the
redirections from "elfgcchack.h" are still included. The symbols will
only be available when building the library inside ADSupport, so we can
only define IN_LIBXML in that case.
---
 supportApp/xml2Src/Makefile          | 1 +
 supportApp/xml2Src/nanohttp_stream.c | 1 -
 2 files changed, 1 insertion(+), 1 deletion(-)

diff --git a/supportApp/xml2Src/Makefile b/supportApp/xml2Src/Makefile
index 1ba2696..1a0c189 100644
--- a/supportApp/xml2Src/Makefile
+++ b/supportApp/xml2Src/Makefile
@@ -123,6 +123,7 @@ ifeq ($(XML2_EXTERNAL),NO)
   LIB_SRCS += xmlstring.c
 
   nanohttp_stream_LIBS += xml2
+  USR_CFLAGS = -DIN_LIBXML
 
 else 
   nanohttp_stream_SYS_LIBS += xml2
diff --git a/supportApp/xml2Src/nanohttp_stream.c b/supportApp/xml2Src/nanohttp_stream.c
index 9d234c1..0433a9e 100644
--- a/supportApp/xml2Src/nanohttp_stream.c
+++ b/supportApp/xml2Src/nanohttp_stream.c
@@ -12,7 +12,6 @@
  */
 
 #define NEED_SOCKETS
-#define IN_LIBXML
 #include "libxml.h"
 
 #ifdef LIBXML_HTTP_ENABLED
